straxen/plugins/acqmon_processing.py

```diff
@@ -0,0 +1,132 @@
+    
+# DDC10 High Energy Veto:
+# 10ms (1e7 ns) - fixed HE veto length in XENON1T DDC10, 
+# in XENONnT it will be calibrated based on the length of large S2 SE tails
+# The start/stop signals for the HEV are generated by the V1495 board
+  
+@export
+@strax.takes_config(
+    strax.Option('max_veto_gap', default = int(5e8),
+                 help = 'Maximum separation between veto stop and start pulses [ns]'),
+    strax.Option('channel_map', track = False, type = immutabledict, 
+                 help = 'immutabledict mapping subdetector to (min, max)'
+                      'channel number.'))
+
+class VetoIntervals(strax.OverlapWindowPlugin):
+    """ Find pairs of veto start and veto stop signals and the veto duration between them
+    busy_*  <= V1495 busy veto for tpc channels
+    he_*    <= V1495 busy veto for high energy tpc channels
+    hev_*   <= DDC10 hardware high energy veto
+    """
+        
+    __version__ ='0.0.23'
+    depends_on = ('aqmon_hits') 
+    provides  = ('veto_intervals')
+    data_kind = ('veto_intervals')
+
+    def infer_dtype(self):
+        dtype = [(('veto start time since unix epoch [ns]', 'time'), np.int64),
+                 (('veto end time since unix epoch [ns]', 'endtime'), np.int64),
+                 (('veto interval [ns]', 'veto_interval'), np.int64),
+                 (('veto signal type', 'veto_type'), np.str('U10'))]
+        return dtype
+    
+    def setup(self):
+        aq_ch_start = 800
+        aqmon_channel_names = ('sum_wf','m_veto_sync',
+                               'hev_stop', 'hev_start', 'he_stop','he_start','busy_stop', 'busy_start')
+        channel_numbers = np.arange(aq_ch_start, aq_ch_start + len(aqmon_channel_names))
+        
+        self.tmp_dtype = strax.time_fields
+        # Keeping ch_range as class attribute for now, maybe will need it for other features
+        self.channel_range = self.config['channel_map']['aqmon']
+        self.channel_map = dict(zip(aqmon_channel_names, channel_numbers))
+        self.veto_names = ['busy_','he_','hev_']
+        return self.channel_map
+    
+    def get_window_size(self):
+        # Give a very wide window
+        return (self.config['max_veto_gap']*100)
+    
+    def compute(self, aqmon_hits): 
+        hits = aqmon_hits
+        
+        res = dict()
+        for i, veto in enumerate(self.veto_names):
+            channels = channel_select(hits, self.channel_map[veto + 'stop'], self.channel_map[veto + 'start'])
+            vetos = self.merge_vetos(channels, gap = self.config['max_veto_gap'],\
+                                           dtype = self.tmp_dtype, t = 0)
+            
+            # Check that we found a veto start time and then populate the resulting dict
+            if np.any(vetos['time']):
+                res['time'] = vetos['time']
+                res['endtime'] = vetos['endtime']
+                res['veto_interval'] = vetos['endtime'] - vetos['time']
```

Comment:
Is not it redundant? You could save 1/3 of the data_type size by removing this field. Computing it should be fairly fast and simple too.

Reply:
I like it, as if we plug the result of this plugin straight into DB it would give me fewer headaches of calculating things in NODIAQ. This data kind was already tiny and I've made it only a fraction of the original one, so data size is also not an issue.

Reply:
Okay, I am fine with it